conda/cli/activate.py

```diff
@@ -24,95 +38,105 @@ def help():
-            import conda.config
-            binpath = binpath_from_arg(os.getenv('CONDA_DEFAULT_ENV'))
-            rootpath = binpath_from_arg(conda.config.root_env_name)
-        except SystemExit:
-            print(os.environ['PATH'])
-            raise
-        # deactivate is the same as activate root (except without setting
-        # CONDA_DEFAULT_ENV or PS1). XXX: The user might want to put the root
-        # env back somewhere in the middle of the PATH, not at the beginning.
-        if rootpath not in os.getenv('PATH').split(os.pathsep):
-            paths = [rootpath]
-        else:
-            paths = []
-        sys.stderr.write("discarding %s from PATH\n" % binpath)
+        path = os.getenv("PATH")
+        if os.getenv("CONDA_ACTIVE_ENV"):
+            binpath = binpath_from_arg(os.getenv('CONDA_ACTIVE_ENV'))
+            if binpath:
+                sys.stderr.write("discarding %s from PATH\n" % pathlist_to_str(binpath))
+            path = path.replace(os.pathsep.join(binpath)+os.pathsep, "", 1)
 
     elif sys.argv[1] == '..checkenv':
         if len(sys.argv) < 3:
             sys.exit("Error: no environment provided.")
         if len(sys.argv) > 3:
             sys.exit("Error: did not expect more than one argument.")
-        binpath = binpath_from_arg(sys.argv[2])
+        if sys.argv[2] == 'root':
+            # no need to check root env and try to install a symlink there
+            sys.exit(0)
+        binpath = binpath_from_arg(sys.argv[2])  # this should throw an error and exit if the env or path can't be found.
         # Make sure an env always has the conda symlink
         try:
-            import conda.config
-            if binpath != join(conda.config.root_dir, 'bin'):
-                import conda.install
-                conda.install.symlink_conda(join(binpath, '..'), conda.config.root_dir)
+            conda.install.symlink_conda(binpath[0], conda.config.root_dir)
         except (IOError, OSError) as e:
-            import errno
             if e.errno == errno.EPERM or e.errno == errno.EACCES:
                 sys.exit("Cannot activate environment {}, do not have write access to write conda symlink".format(sys.argv[2]))
             raise
         sys.exit(0)
 
+    elif sys.argv[1] == '..setps1':
+        # path is a bit of a misnomer here.  It is the prompt setting.  However, it is returned
+        #    below by printing.  That is why it is named "path"
+        path = sys.argv[3]
+        if not path:
+            if on_win:
+                path = os.getenv("PROMPT", "$P$G")
+            else:
+                # zsh uses prompt.  If it exists, prefer it.
+                path = os.getenv("PROMPT")
+                # fall back to bash default
+                if not path:
+                    path = os.getenv("PS1")
+        # strip off previous prefix, if any:
+        path = re.sub("^\(.*\)\ ", "", path)
+        env_path = sys.argv[2]
+        if conda.config.changeps1 and env_path:
```

Comment:
I'm very confused by this - it works fine for me:

```
C:\Users\builder>activate py35
prepending C:\Users\builder\MC_Main\envs\py35 and C:\Users\builder\MC_Main\envs\py35\Scripts and C:\Users\builder\MC_Main\envs\py35\Library\bin to PATH

(py35) C:\Users\builder>deactivate
discarding C:\Users\builder\MC_Main\envs\py35 and C:\Users\builder\MC_Main\envs\py35\Scripts and C:\Users\builder\MC_Main\envs\py35\Library\bin from PATH

C:\Users\builder>
```

How can I replicate your system?  Clink?  Cmder?  Do you have any other setup that I can install and test with?